File: StreamlineAcademyWebApi/StreamlineAcademy.Domain/Entities/CourseResource.cs
using StreamlineAcademy.Domain.Enums;
using StreamlineAcademy.Domain.Shared;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace StreamlineAcademy.Domain.Entities
{
    public class CourseResource:BaseModel
    {
        public string? Name { get; set; }
        public string? Description { get; set; }
        public  Enums.CourseResourceType? Type { get; set; }
        public string? FilePath { get; set; } 
        public Guid CourseId { get; set; }
        [ForeignKey(nameof(CourseId))]
        public Course? Course { get; set; }

    }
}

File: StreamlineAcademyWebApp/Sl-Academy-App/src/app/Models/CourseResource/CourseResource.ts
import { CourseResourceType } from "../../Enums/courseresourcetype";

export class CourseResource{
    Name?:string;
    Description?:string;
    Type?:CourseResourceType
    CourseId?:string=''
    File?:File;
}

export class CourseResourceResponse{
    id?:string;
    name?:string;
    description?:string;
    filePath?:string;
    type?:CourseResourceType;
    courseId?:string=''
    courseName?:string;
    isActive?:boolean;
    File?:File
}

export class UpdateCourseResource{
    Id?:string;
    Name?:string;
    Description?:string;
    Type?:CourseResourceType;
    CourseId?:string;
    File?:File;
}